wandb/sdk/launch/runner/gcp_vertex.py

```diff
@@ -0,0 +1,244 @@
+        )
+
+        validate_docker_installation()
+        synchronous: bool = self.backend_config[PROJECT_SYNCHRONOUS]
+        docker_args: Dict[str, Any] = self.backend_config[PROJECT_DOCKER_ARGS]
+        if docker_args:
+            wandb.termwarn(
+                "Docker args are not supported for GCP. Not using docker args"
+            )
+
+        entry_point = launch_project.get_single_entry_point()
+
+        entry_cmd = entry_point.command
+        copy_code = True
+        if launch_project.docker_image:
+            pull_docker_image(launch_project.docker_image)
+            copy_code = False
+        else:
+            # TODO: potentially pull the base_image
+            if not docker_image_exists(launch_project.base_image):
+                if generate_docker_base_image(launch_project, entry_cmd) is None:
+                    raise LaunchError("Unable to build base image")
+            else:
+                wandb.termlog(
+                    "Using existing base image: {}".format(launch_project.base_image)
+                )
+
+        command_args = []
+        command_args += get_entry_point_command(
+            entry_point, launch_project.override_args
+        )
+
+        container_inspect = docker_image_inspect(launch_project.base_image)
+        container_workdir = container_inspect["ContainerConfig"].get("WorkingDir", "/")
+        container_env: List[str] = container_inspect["ContainerConfig"]["Env"]
+
+        if launch_project.docker_image is None or launch_project.build_image:
+            image_uri = construct_gcp_image_uri(
+                launch_project, gcp_artifact_repo, gcp_project, gcp_docker_host,
+            )
+            image = build_docker_image_if_needed(
+                launch_project=launch_project,
+                api=self._api,
+                copy_code=copy_code,
+                workdir=container_workdir,
+                container_env=container_env,
+                runner_type="gcp-vertex",
+                image_uri=image_uri,
+                command_args=command_args,
+            )
+        else:
+            image = launch_project.docker_image
+
+        docker_push(image)  # todo: when aws pr is merged, use docker python tooling
+
+        if self.backend_config.get("runQueueItemId"):
+            try:
+                self._api.ack_run_queue_item(
+                    self.backend_config["runQueueItemId"], launch_project.run_id
+                )
+            except CommError:
+                wandb.termerror(
+                    "Error acking run queue item. Item lease may have ended or another process may have acked it."
+                )
+                return None
+
+        job = aiplatform.CustomContainerTrainingJob(
+            display_name=gcp_training_job_name, container_uri=image,
+        )
+        submitted_run = VertexSubmittedRun(job)
+
+        # todo: support gcp dataset?
```

Comment:
In Sagemaker you can set the data input as a file/stream and a link to an s3 bucket. This is set by the user as a resource arg and used in the job creation API. Does Vertex have a similar API?